src/query/shorthand.ts

```diff
@@ -409,23 +421,31 @@ export namespace shorthandParser {
 
     let partParams = fieldDefPart[2];
     let closingBraceIndex = 0;
+    let i = 0;
 
-    // FIXME: don't use a loop
-    // FIXME: separate maxbins from nestedPropertyParent and support other bin properties
-    for (let nestedPropertyParent of ['maxbins', 'scale', 'sort', 'axis', 'legend']) {
-      let nestedPropertyParentIndex = partParams.indexOf(nestedPropertyParent, closingBraceIndex);
-      if (nestedPropertyParentIndex !== -1) {
+    while (i < partParams.length) {
+      let propEqualSignIndex = partParams.indexOf('=', i);
+
+      if (propEqualSignIndex !== -1) {
+        let nestedPropertyParent = partParams.substring(i, propEqualSignIndex);
```

Comment:
Maybe rename this to `prop`?  In a way, this could be either a nested property's parent or a child (like for bin properties).